include/deal.II/lac/la_parallel_vector.templates.h

```diff
@@ -925,21 +930,27 @@ namespace LinearAlgebra
     void
     Vector<Number, MemorySpaceType>::zero_out_ghost_values() const
     {
-      if (data.values != nullptr)
-        std::fill_n(data.values.get() + partitioner->locally_owned_size(),
-                    partitioner->n_ghost_indices(),
-                    Number());
-#ifdef DEAL_II_COMPILER_CUDA_AWARE
-      if (data.values_dev != nullptr)
+      if (data.values.size() != 0)
         {
-          const cudaError_t cuda_error_code =
-            cudaMemset(data.values_dev.get() +
-                         partitioner->locally_owned_size(),
-                       0,
-                       partitioner->n_ghost_indices() * sizeof(Number));
-          AssertCuda(cuda_error_code);
-        }
+#ifdef DEAL_II_COMPILER_CUDA_AWARE
+          if (std::is_same<MemorySpaceType, MemorySpace::CUDA>::value)
+            {
+              const cudaError_t cuda_error_code =
+                cudaMemset(data.values.data() +
+                             partitioner->locally_owned_size(),
+                           0,
+                           partitioner->n_ghost_indices() * sizeof(Number));
+              AssertCuda(cuda_error_code);
+            }
+          else
 #endif
+            {
+              std::fill_n(data.values.data() +
+                            partitioner->locally_owned_size(),
+                          partitioner->n_ghost_indices(),
+                          Number());
+            }
+        }
```

Comment:
All of this can be cleaned up but we can do it in a different PR.

Reply:
Yes, see https://github.com/dealii/dealii/pull/14537/files#diff-f2ee1054edd7dc30feb86f7331a47a9a0c50f82de59bb66c61d8de73f062a96f.